frontend/src/pages/pipelines/global/GlobalPipelineCoreLoader.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { Navigate, Outlet, useParams } from 'react-router-dom';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import { byName, ProjectsContext } from '~/concepts/projects/ProjectsContext';
+import PipelineCoreNoProjects from '~/pages/pipelines/global/PipelineCoreNoProjects';
+import PipelineCoreProjectSelector from '~/pages/pipelines/global/PipelineCoreProjectSelector';
+import { PipelineContextProvider } from '~/concepts/pipelines/context';
+import InvalidProject from '~/concepts/projects/InvalidProject';
+
+type ApplicationPageProps = React.ComponentProps<typeof ApplicationsPage>;
+type EmptyStateProps = 'emptyStatePage' | 'empty';
+
+type GlobalPipelineCoreLoaderProps = {
+  getInvalidRedirectPath: (namespace: string) => string;
+} & Omit<
+  ApplicationPageProps,
+  'loaded' | 'headerContent' | 'provideChildrenPadding' | EmptyStateProps
+>;
+
+type ApplicationPageRenderState = Pick<ApplicationPageProps, EmptyStateProps>;
+
+const GlobalPipelineCoreLoader: React.FC<GlobalPipelineCoreLoaderProps> = ({
+  getInvalidRedirectPath,
+  ...applicationPageProps
+}) => {
+  const { namespace } = useParams<{ namespace: string }>();
+  const { projects } = React.useContext(ProjectsContext);
+
+  let renderStateProps: ApplicationPageRenderState & { children?: React.ReactNode };
+  if (projects.length === 0) {
+    renderStateProps = {
+      empty: true,
+      emptyStatePage: <PipelineCoreNoProjects />,
+    };
+  } else {
+    if (namespace) {
+      const foundProject = projects.find(byName(namespace));
+      if (foundProject) {
+        // Render the content
+        return (
+          <PipelineContextProvider namespace={namespace}>
```

Comment:
This component is a glorified context provider wrapper -- but also handles the "if no projects" situation. Allows for the pipeline pages to be setup for nested namespace routes.